src/NHibernate/Proxy/AbstractProxyFactory.cs

```diff
@@ -19,11 +19,7 @@ public abstract class AbstractProxyFactory: IProxyFactory
 		protected virtual MethodInfo SetIdentifierMethod { get; private set; }
 		protected virtual IAbstractComponentType ComponentIdType { get; private set; }
 		protected virtual bool OverridesEquals { get; set; }
-
-		protected bool IsClassProxy
-		{
-			get { return Interfaces.Length == 1; }
-		}
+		protected internal bool IsClassProxy { get; internal set; }
```

Comment:
Maybe should be true by default?

Reply:
And why internal properties in public class? I think it should stay simply protected.

Reply:
> why internal properties in public class? I think it should stay simply protected.

The value is set outside the class so property is `protected internal`. I do not want consumers to change the value so setter is `internal`.

Reply:
Ok. I just find it strange that we allow to create custom proxy based on this class but doesn't really allow to adjust this property in any way considering that this value must be set outside.

Reply:
I did not want to change the behaviour. It was calculated property, so I moved calculation outside.